pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -550,18 +550,55 @@ func (service *SubnetPortService) AllocatePortFromSubnet(subnet *model.VpcSubnet
 
 	info.lock.Lock()
 	defer info.lock.Unlock()
+
+	dhcpMode := "DHCP_DEACTIVATED"
+	subnetInfo, _ := servicecommon.ParseVPCResourcePath(*subnet.Path)
+	if subnet.SubnetDhcpConfig != nil && subnet.SubnetDhcpConfig.Mode != nil {
+		dhcpMode = *subnet.SubnetDhcpConfig.Mode
+	}
+	// For DHCP Server mode Subnet, get total IPs from DHCP IP Pool from NSX each time
+	// since user might update reservedIPRanges for the subnet and it impacts the DHCP Pool size
+	if dhcpMode == "DHCP_SERVER" {
+		dhcpServerStats, err := service.NSXClient.StatsClient.Get(subnetInfo.OrgID, subnetInfo.ProjectID, subnetInfo.VPCID, subnetInfo.ID, nil, nil, nil, nil, nil, nil, nil)
+		if err != nil {
+			log.Error(err, "Failed to get Subnet dhcp-server-config stats", "Subnet", *subnet.Path)
+			return false
+		}
+		info.totalIP = int(*dhcpServerStats.IpPoolStats[0].PoolSize)
```

Comment:
Maybe also check the nil pointer before dereference it for safety?

Besides, would you know if IpPoolStats can have length more than 1 and in that case if we need to check a specific ippoolstates or add them up?

Reply:
I checked currently in NSX only 1 IpPoolStats for the Subnet.